primap2/_overview.py

```diff
@@ -0,0 +1,98 @@
+"""Summarization and descriptive statistics functions to get an overview of a data
+set."""
+
+import typing
+
+import pandas as pd
+
+from . import _accessor_base
+
+
+class DataArrayOverviewAccessor(_accessor_base.BaseDataArrayAccessor):
+    def coverage(
+        self, coord_x: typing.Hashable, coord_y: typing.Hashable
+    ) -> pd.DataFrame:
+        """Summarize how many data points exist for a coordinate combination.
+
+        For each combinations of values in the given coordinates, count the number of
+        non-NaN data points in the array. The result is returned as a two-dimensional
+        pandas DataFrame.
+
+        Parameters
+        ----------
+        coord_x: str
+            Name or alias of the first coordinate, which will be the x-coordinate, i.e.
+            the columns of the result.
+        coord_y: str
+            Name or alias of the second coordinate, which will be the x-coordinate, i.e.
+            the rows of the result.
+
+        Returns
+        -------
+        coverage: pandas.DataFrame
+            Two-dimensional dataframe summarizing the number of non-NaN data points
+            for each combination of value from the x and y coordinate.
+        """
+        coord_x = self._da.pr.dim_alias_translations.get(coord_x, coord_x)
+        coord_y = self._da.pr.dim_alias_translations.get(coord_y, coord_y)
+        return (
+            self._da.count(dim=set(self._da.dims) - {coord_x, coord_y})
+            .transpose(coord_y, coord_x)
+            .to_dataframe("coverage")["coverage"]
+            .unstack()
+        )
+
+
+class DatasetOverviewAccessor(_accessor_base.BaseDatasetAccessor):
```

Comment:
The short names "cat" and "scen" are an implementation detail in the attrs, their public short names are "category" and "scenario" (i.e. without the terminology).

The behaviour of coverage() here is consistent with the behaviour of the `pr.loc[]` and `pr[]` getitem functions and I'd like to keep it consistent; however, it would of course be possible to add short aliases `scen` and `cat` everywhere and make them part of the public API. I think it could save some typing, but it can also be a bit non-obvious that `ds.pr["cat"] == ds.pr["category"] == ds["category (IPCC2006)"]`.

Reply:
And the second thing is a better error message when a non-existing dimension is used. I'll add that.

Reply:
OK, then I misunderstood how the attrs are used. I think it's a bit confusing (at least to me as proven by example) that there are three names for one coordinate and one that is visible in the data structure in attrs can not be used for data access while a non-visible one can be used. Can you explain why it's  `scen` and `cat` in attrs if that can not be used in `.loc` statements?

Reply:
Nope, I can't explain it. I concur that it is confusing. It is not much work to add `scen` and `cat` aliases for `.loc` statements (and everywhere else), so I'll do that.